src/Microsoft.Dnx.Tooling/Publish/PublishProject.cs

```diff
@@ -550,11 +550,16 @@ private bool GenerateWebConfigFileForWwwRootOut(PublishRoot root, Runtime.Projec
 
             root.Reports.Information.WriteLine($"Using command '{command}' as the entry point for web.config.");
 
+            var azurePublishValue = Environment.GetEnvironmentVariable("DNU_PUBLISH_AZURE");
```

Comment:
Instead of relying on a new env variable which requires a Kudu change, I suggest using an existing one to detect if you're running in Azure Web App. e.g. just check if `WEBSITE_SITE_NAME` is defined.